### DISCUSSION TITLE
Application Integration through OpenID Connect / OAuth 2.0

### DISCUSSION THREAD

**Discussion by FerdinandvHagen** [teamhanko/hanko#860] [Ideas]
I would like to propose using the OIDC/OAuth 2.0 flow for the session flow / integration with applications instead of the current JWT based flow - or as a basis for the development of a new session flow as announced in this [comment](https://github.com/teamhanko/hanko/discussions/246#discussioncomment-4290364). In contrast to #782 this is not about adding general OIDC/OAuth 2.0 capabilities to login users - but use the OIDC/OAuth 2.0 as the underlying protocol flow for applications to authenticate users through Hanko. For those familiar with OIDC/OAuth 2.0 - essentially making Hanko an OIDC OpenID Provider under the OIDC spec.

[Zitadel](https://zitadel.com/) uses this process and it makes for very simple integration with applications as most languages/frameworks support OIDC/OAuth2.0 out of the box. An explanation of the process and how Zitadel handles this can be found [here](https://zitadel.com/docs/guides/integrate/login-users).

I would expect that any custom session management / application integration process will eventually become very similar to OIDC/OAuth2.0 anyways - and as already discussed in #246 the current usage of a JWT for session management is at least debatable - if not discouraged.

To make implementation easy, Zitadel has open-sourced the OIDC/OAuth2.0 server implementation (GoLang). An example on how to use the library can be found [here](https://github.com/zitadel/oidc/tree/main/example/server)).

**Comment by mfechtner** [comment]
Hi is there progress on this? I have seen one or two oauth topics on the roadmap.
I might be wrong, but it does not look like it will be implemented soon in hanko.

**Reply by FlxMgdnz** [reply]
We’re revisiting the topic right now, and I can imagine that we’ll start the actual implementation fairly soon.

**Comment by oluwadele** [comment]
> I would like to propose using the OIDC/OAuth 2.0 flow for the session flow / integration with applications instead of the current JWT based flow - or as a basis for the development of a new session flow as announced in this [comment](https://github.com/teamhanko/hanko/discussions/246#discussioncomment-4290364). In contrast to #782 this is not about adding general OIDC/OAuth 2.0 capabilities to login users - but use the OIDC/OAuth 2.0 as the underlying protocol flow for applications to authenticate users through Hanko. For those familiar with OIDC/OAuth 2.0 - essentially making Hanko an OIDC OpenID Provider under the OIDC spec.
> 
> [Zitadel](https://zitadel.com/) uses this process and it makes for very simple integration with applications as most languages/frameworks support OIDC/OAuth2.0 out of the box. An explanation of the process and how Zitadel handles this can be found [here](https://zitadel.com/docs/guides/integrate/login-users).
> 
> I would expect that any custom session management / application integration process will eventually become very similar to OIDC/OAuth2.0 anyways - and as already discussed in #246 the current usage of a JWT for session management is at least debatable - if not discouraged.
> 
> To make implementation easy, Zitadel has open-sourced the OIDC/OAuth2.0 server implementation (GoLang). An example on how to use the library can be found [here](https://github.com/zitadel/oidc/tree/main/example/server)).

Awesome
